bundles/binding/org.openhab.binding.fritzboxtr064/src/main/java/org/openhab/binding/fritzboxtr064/internal/Helper.java

```diff
@@ -34,20 +35,26 @@
 
     /***
      * Helper method which converts XML Document into pretty formatted string
-     * 
+     *
      * @param doc to convert
      * @return converted XML as String
      */
     public static String documentToString(Document doc) {
+
         String strMsg = "";
-        OutputFormat format = new OutputFormat(doc);
-        format.setIndenting(true);
-        ByteArrayOutputStream baos = new ByteArrayOutputStream();
-        XMLSerializer serializer = new XMLSerializer(baos, format);
         try {
-            serializer.serialize(doc);
-            strMsg = baos.toString("UTF-8");
-        } catch (IOException e) {
+            DOMImplementation domImpl = doc.getImplementation();
+            DOMImplementationLS domImplLS = (DOMImplementationLS) domImpl.getFeature("LS", "3.0");
+            LSSerializer lsSerializer = domImplLS.createLSSerializer();
+            lsSerializer.getDomConfig().setParameter("format-pretty-print", true);
+
+            Writer stringWriter = new StringWriter();
+            LSOutput lsOutput = domImplLS.createLSOutput();
+            lsOutput.setEncoding("UTF-8");
+            lsOutput.setCharacterStream(stringWriter);
+            lsSerializer.write(doc, lsOutput);
+            strMsg = stringWriter.toString();
+        } catch (Exception e) {
             e.printStackTrace();
```

Comment:
Could you replace this and any occurrences of printStackTrace with logger.warn in the code (since you are making other fixes)? Bindings should not print to stdout or stderr but instead use log4j.